### PR TITLE
Fix miniMQTT compatibility and native_networking examples

### PR DESCRIPTION
**Pull request by askpatrickw**
- added ssl_context to creation of MQTT() <-- This is the main fix
- changed minimqtt import to match minimqtt samples
- Changed secrets['key'] to secrets['device_sas_key'] because we want the IoT Central **DEVICE** SAS Key not the API Key, or the IoT Central Application SAS key
- Moved existing examples into examples/esp32spi subfolder, inline with minimqtt samples
- Created examples/native_networking folder
- Created and tested native_networking examples. All tested on Metro ESP32-S2 and passed.
- updated .gitignore with full set of "ignores" as recommended by VSCode's extension
- Python coding style changes as per sourcery
- Updated docs: added esp32spi and native networking imports, updated the dependencies in the __init__.py docstring
- Added blank line after description before params in docstring to fix layout issues where parameters were not formatted properly in Sphinx HTML output
- Fixed Doc Build Failure as FoamyGuy suggests in the open issue: https://github.com/adafruit/Adafruit_CircuitPython_AzureIoT/issues/38

ToDo:

- [ ] Test ESP32SPI examples.
NOTE: I do not have an ESP32SPI board, if someone else could try at least the simpletest examples, that would be awesome.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -203,7 +203,7 @@ To use Azure IoT Central, you will need to create an Azure IoT Central app, crea
 
     from adafruit_azureiot import IoTCentralDevice
 
-    device = IoTCentralDevice(wifi, secrets["id_scope"], secrets["device_id"], secrets["key"])
+    device = IoTCentralDevice(wifi, secrets["id_scope"], secrets["device_id"], secrets["device_device_sas_key"])
```
Comment (brentru) [inline README.rst:206]: Agree with this change, pls. tag me when this is read for final review as the guide's documentation will need to be modified as well.

**Review by brentru — COMMENTED** [review]
> I removed the neopixel stuff from simple examples. I find these "extras" confusing when learning a new library.

I'd prefer to keep this in, modify the behavior of the examples as little as possible.



> I am using adafruit_requests and the adafruit_io time api to set time. Is this cool as a general approach?

Perfectly fine

File: README.rst (hunk 1 above)
Comment (askpatrickw) [reply README.rst:206]: > > I removed the neopixel stuff from simple examples. I find these "extras" confusing when learning a new library.
> 
> I'd prefer to keep this in, modify the behavior of the examples as little as possible.

I should clarify. I shouldn't have said I'm "removing" anything as I'm not changing the ESP32SPI samples. They use the wifi manager which takes a status light parameter.

`wifi = adafruit_esp32spi_wifimanager.ESPSPI_WiFiManager(esp, secrets, status_light)`

For boards with native networking, there is not a similar status_light to be passed.

`wifi.radio.connect(secrets["ssid"], secrets["password"])`

Comment (askpatrickw) [reply README.rst:206]: Also the readme is a typo it will be `device_sas_key` not `device_device_sas_key`. 

I'll definitely do a pass through the docs as well.

**Comment by brentru** [comment]
@askpatrickw  Did you test this on the PyPortal (or any "AirLift" hardware)?

**Comment by askpatrickw** [comment]
No. Only on a MetroS2. I don't have those other boards.

File: README.rst [hunk 2]
```diff
@@ -79,7 +79,7 @@ To use this library, you will need to create an ESP32_SPI WifiManager, connected
 
 Native Networking
 =================
-To use this library, with boards that have native networking support, you need to be connected to a network. You will also need to set the current time, as this is used to generate time-based authentication keys. One way to do this is by using the `Adafruit IoT Time Service <https://io.adafruit.com/api/docs/#time>`_ via the `Requests library <https://github.com/adafruit/Adafruit_CircuitPython_Requests/>_` with the following code:
+To use this library, with boards that have native networking support, you need to be connected to a network. You will also need to set the current time, as this is used to generate time-based authentication keys. One way to do this is by using the `Adafruit IoT Time Service <https://io.adafruit.com/api/docs/#time>`_ via the `Adafruit Requests library <https://github.com/adafruit/Adafruit_CircuitPython_Requests>`_ with the following code:
```
Comment (brentru) [inline README.rst:82]: Please change

> Adafruit IoT Time Service

to 


Adafruit IO Time API

File: adafruit_azureiot/__init__.py [hunk 3]
```diff
@@ -19,15 +19,16 @@
 * Adafruit CircuitPython firmware for the supported boards:
   https://github.com/adafruit/circuitpython/releases
 
-**ESP32SPI Peripheral Networking**
+**With ESP32SPI Peripheral Networking**
```
Comment (brentru) [inline adafruit_azureiot/__init__.py:22]: Change to: "ESP32 AirLift Networking", it has name recognition across our documentation on Learn.

File: requirements.txt [hunk 4]
```diff
@@ -4,5 +4,4 @@
 
 Adafruit-Blinka
 Adafruit-CircuitPython-miniMQTT
-Adafruit-CircuitPython-Requests
```
Comment (brentru) [inline requirements.txt:7]: Why is this removed? Don't we need requests?

**Review by brentru — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 4 above)
Comment (askpatrickw) [reply requirements.txt:7]: @brentru The examples need requests, but the library does not.

File: README.rst (hunk 2 above)
Comment (askpatrickw) [reply README.rst:82]: I had that originally too, then I went to get the link and it says service not API.

https://io.adafruit.com/api/docs/#time

image

**Comment by askpatrickw** [comment]
OK @brentru I think I'm done. I can provide Azure API Keys for anyone who can test on an ESP32 Airlift.

**Review by brentru — APPROVED** [review]
